share/mk/bsd.cpu.mk

```diff
@@ -418,13 +418,24 @@ MACHINE_ABI+=	hard-float
 .endif
 .if (${MACHINE_ARCH:Maarch64*c*} || ${MACHINE_ARCH:Mriscv*c*})
 MACHINE_ABI+=	purecap
-.endif
+.else
 # Currently all 64-bit architectures include 64 in their name (see arch(7)).
 .if ${MACHINE_ARCH:M*64*}
 MACHINE_ABI+=	ptr64
 .else
 MACHINE_ABI+=	ptr32
 .endif
+.endif
+.if ${MACHINE_ARCH:M*64*}
+MACHINE_ABI+=	long64
+.else
+MACHINE_ABI+=	long32
+.endif
+.if ${MACHINE_ARCH:Mi386}
```

Comment:
==?

Reply:
Are you suggesting:
```suggestion
.if ${MACHINE_ARCH} == i386
```

Reply:
Yes; using :M for non-globs feels a bit odd